lightning/src/ln/channelmanager.rs

```diff
@@ -3400,7 +3400,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			let short_to_id = &mut channel_state.short_to_id;
 			let pending_msg_events = &mut channel_state.pending_msg_events;
 			channel_state.by_id.retain(|_,  v| {
-				if v.block_disconnected(header) {
+				if v.block_disconnected(header, new_height) {
```

Comment:
Shouldn't this be the old height of the disconnected block?

Reply:
Hmm, I dont think so? Thew new argument is passed into `update_best_block` in channel and handled the same as a new block (and the header part ignored, except to get the time).

Reply:
Note `new_height` is calculated by fetching the height from the local `latest_block_height` variable, not from the user. In later commits, that variable is checked against what the user passes in.